### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by raphaelspies**
private routes, active sessions, and all the fun you wanted to have but were too much of a teapot to contemplate

### CODE REVIEW

File: client/src/components/PrivateRoute.jsx [hunk 1]
```diff
@@ -1,27 +1,55 @@
...
-// }
 
-export default function PrivateRoute({ component: Component, ...rest }) {
+export default function PrivateRoute({ children, ...rest }) {
   const userData = useSelector((store) => store.currentUserReducer.userData);
-  console.log(userData)
-  // const { currentUser } = useAuth();
+
+  function tellMeUser() {
+    console.log(userData);
```
Comment (dchang1010) [inline client/src/components/PrivateRoute.jsx]: here

File: client/src/components/PrivateRoute.jsx [hunk 2]
```diff
@@ -1,27 +1,55 @@
...
+  }
+
+  function checkForSession() {
+    console.log(document.cookie);
```
Comment (dchang1010) [inline client/src/components/PrivateRoute.jsx:13]: here

File: server/controllers/userCtrl.js [hunk 3]
```diff
@@ -108,6 +108,7 @@ const userControllers = {
 
       db.query(queryStr)
         .then((data) => {
+          console.log(data.rows[0])
```
Comment (dchang1010) [inline server/controllers/userCtrl.js]: here

File: server/controllers/userCtrl.js [hunk 4]
```diff
@@ -338,7 +339,9 @@ const userControllers = {
           res.status(404).send('error: password does not match');
         } else {
           //return session
-          req.session.user_Id = user_id;
+          console.log("user_id:", user_id)
```
Comment (dchang1010) [inline server/controllers/userCtrl.js]: here

File: server/controllers/userCtrl.js [hunk 5]
```diff
@@ -350,6 +353,17 @@ const userControllers = {
       });
   },
   // *************************************************************
+  authenticateSession: (req, res) => {
+    console.log(req.session.user_id, "<---------- this is the associated uid");
```
Comment (dchang1010) [inline server/controllers/userCtrl.js]: here

File: server/routes/login.js [hunk 6]
```diff
@@ -26,11 +26,12 @@ login.post('/',
         errors: errors.array(),
       });
     }
+    console.log(req.body);
```
Comment (dchang1010) [inline server/routes/login.js:28]: here

File: client/src/components/PrivateRoute.jsx [hunk 7]
```diff
@@ -2,37 +2,43 @@ import React, { useContext, createContext, useState, useEffect } from "react";
...
-
+import axios from 'axios';
 
 export default function PrivateRoute({ children, ...rest }) {
   const userData = useSelector((store) => store.currentUserReducer.userData);
+  const dispatch = useDispatch();
 
   function tellMeUser() {
-    console.log(userData);
+    console.log('tell me user: ', userData);
```
Comment (dchang1010) [inline client/src/components/PrivateRoute.jsx]: here

**Review by ajmunoz411 — APPROVED** [review]

**Review by dchang1010 — APPROVED** [review]

**Pull request merged**